Errors building with Visual C++

Thank you for your work on this project.  I look forward to using it.

I'm trying to use version 3.2.2 with Visual C++ inside Visual Studio Community 2019 version 16.6.0.  Here are several errors:

```
ConsoleApplication2\tinyutf8.h(3202): error C4703: potentially uninitialized local pointer variable 'app_lut_base_ptr' used
ConsoleApplication2\tinyutf8.h(3198): error C4703: potentially uninitialized local pointer variable 'old_lut_base_ptr' used
ConsoleApplication2\tinyutf8.h(3533): error C4703: potentially uninitialized local pointer variable 'str_lut_base_ptr' used
ConsoleApplication2\tinyutf8.h(3484): error C4703: potentially uninitialized local pointer variable 'old_lut_base_ptr' used
ConsoleApplication2\tinyutf8.h(3932): error C4703: potentially uninitialized local pointer variable 'repl_lut_base_ptr' used
ConsoleApplication2\tinyutf8.h(3880): error C4703: potentially uninitialized local pointer variable 'old_lut_base_ptr' used
```

Any tips for using it in Visual C++?

--- a/.travis.yml
+++ b/.travis.yml
@@ -11,3 +11,8 @@ compiler:
 
 script:
   - $CXX -std=c++11 -Wall -Wextra -pedantic -Wno-attributes -I. test/test.cpp
+  
+jobs:
+  exclude:
+  - os: windows
+    compiler: clang

--- a/.travis.yml
+++ b/.travis.yml
@@ -3,6 +3,7 @@ language: cpp
 os:
   - linux
   - osx
+  - windows
 
 compiler:
   - gcc

--- a/tinyutf8.h
+++ b/tinyutf8.h
@@ -4438,6 +4438,5 @@ std::istream& operator>>( std::istream& stream , utf8_string& str ){
 #elif defined (_MSC_VER)
 #pragma warning(pop)
 #endif
-#endif
 
 #endif // _TINY_UTF8_H_

--- a/tinyutf8.h
+++ b/tinyutf8.h
@@ -52,10 +52,16 @@
 	#endif
 #endif
 
-//! Remove -Wmaybe-uninitialized, since it is wrong for all cases in this file
-#if defined (__GNUC__)
+//! Remove Warning "-Wmaybe-uninitialized" (GCC/Clang) resp. "C4703" (MSVC), since it is wrong for all cases in this file
+#if defined (__clang__)
+#pragma clang diagnostic ignored "-Wmaybe-uninitialized"
+#pragma clang diagnostic push
+#elif defined (__GNUC__)
 #pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
 #pragma GCC diagnostic push
+#elif defined (_MSC_VER)
+#pragma warning(push)
+#pragma warning(disable:4703)
 #endif
 
 namespace tiny_utf8_detail
@@ -4425,8 +4431,13 @@ std::istream& operator>>( std::istream& stream , utf8_string& str ){
 
 #endif // TINY_UTF8_FORWARD_DECLARE_ONLY
 
-#if defined (__GNUC__)
+#if defined (__clang__)
+#pragma clang diagnostic pop
+#elif defined (__GNUC__)
 #pragma GCC diagnostic pop
+#elif defined (_MSC_VER)
+#pragma warning(pop)
+#endif
 #endif
 
 #endif // _TINY_UTF8_H_

